lib/trivial.nix

```diff
@@ -467,6 +471,41 @@ rec {
     then v
     else k: v;
 
+  /*
+    Make a function lazy in its arguments.
+
+    Normally when you call a function like `{ a, b }: { r = foo a b; }`,
+    the attribute names of the argument must be evaluated before the evaluator
+    can return a the (partially evaluated) function body (`{ r = ...; }`).
+
+    `lazyFunction` uses reflection on the passed in function and it only works
+    for functions defined with a "set pattern".
+
+    It makes a function like `{ a, b }: { r = foo a b; }` behave like
+    `args: { r = foo args.a args.b; }`.
```

Comment:
```suggestion
    In other words, it makes a function like `{ a, b }: { r = foo a b; }` behave like
    `args: { r = foo args.a args.b; }` instead.
```

Reply:
I've changed this to show the "generated wrapper" instead. I think that helps more with understanding.